components/admin/lessons/AdminLessonInfo.tsx

```diff
@@ -152,14 +153,15 @@ export const AdminLessonInfo: React.FC<LessonInfoProps> = ({
   selectedLesson
 }) => {
   // true when user clicks on `create new lesson` button
-  if (lessons && selectedLesson === lessons.length) {
+  if (!lessons || selectedLesson === lessons.length) {
     return <NewLesson setLessons={setLessons} />
   }
 
   // set currently selected lesson
-  const lesson = lessons && lessons[selectedLesson]
+  const lesson = lessons[selectedLesson] ? lessons[selectedLesson] : lessons[0]
 
-  const lessonId = parseInt(lesson ? lesson.id + '' : '')
+  //TO-DO change Lesson type to include non-nullable Int id field
+  const lessonId = parseInt(lesson.id + '')
```

Comment:
Our grapql types strike again. According to our type system `lesson.id` can be null but every lesson mutation requires id. So this component will render everything correctly but nothing will work. 

Ideally lesson.id should be non-nullable int.